src/streaming/zarr.stream.cpp

```diff
@@ -543,20 +543,8 @@ ZarrStream_s::create_store_()
     if (is_s3_acquisition_()) {
         // spin up S3 connection pool
         try {
-            if (s3_settings_->region.has_value()) {
-                s3_connection_pool_ = std::make_shared<zarr::S3ConnectionPool>(
-                  std::thread::hardware_concurrency(),
-                  s3_settings_->endpoint,
-                  s3_settings_->access_key_id,
-                  s3_settings_->secret_access_key,
-                  *s3_settings_->region);
-            } else {
-                s3_connection_pool_ = std::make_shared<zarr::S3ConnectionPool>(
-                  std::thread::hardware_concurrency(),
-                  s3_settings_->endpoint,
-                  s3_settings_->access_key_id,
-                  s3_settings_->secret_access_key);
-            }
+            s3_connection_pool_ = std::make_shared<zarr::S3ConnectionPool>(
+              std::thread::hardware_concurrency(), *s3_settings_);
```

Comment:
shouldn't this follow the thread limit that a user sets?

If so, that doesn't have to get fixed in this PR; just create an issue.